back/server/modules/battleshipRound.js

```diff
@@ -180,6 +187,27 @@ class BattleshipRound {
         return {x:x, y:y}
     }
 
+    targetShipTiles(opMap) {
+        // If the target list is empty, we need to scan the grid and populate it
```

Comment:
Let's invert the order of these. So:

if (this.aiTargetList.Length > 0) {
  return this.aiTargetList.shift();
}

THEN generate the list and return the first value. That removes the need for the length check of 0 and some of the nesting.